src/lib/containers/discussion_forum/ForumTable.tsx

```diff
@@ -24,11 +34,33 @@ export const ForumTable: React.FC<ForumTableProps> = ({
   limit,
   filter,
 }) => {
+  const { accessToken } = useSynapseContext()
   const [sort, setSort] = useState<DiscussionThreadOrder>(
     DiscussionThreadOrder.PINNED_AND_LAST_ACTIVITY,
   )
+  const [subscribed, setSubscribed] = useState<Subscription>()
   const [isAscending, setIsAscending] = useState(false)
 
+  async function getSubscribe() {
+    let subscriptionRequest: SubscriptionRequest = {
+      objectType: SubscriptionObjectType.FORUM,
+      idList: [forumId],
+      sortByType: SortByType.OBJECT_ID,
+      sortDirection: Direction.ASC,
+    }
+    let subscriptionList = await SynapseClient.postSubscriptionList(
+      accessToken,
+      subscriptionRequest,
+    )
+    if (subscriptionList.totalNumberOfResults > 0) {
+      let currentSubscription = subscriptionList.results[0]
+      setSubscribed(currentSubscription)
+    }
+  }
+  useEffect(() => {
+    getSubscribe()
+  }, [accessToken])
```

Comment:
Either define the function in the effect or outside of the component. Then ESLint will tell you what dependencies you're missing in the effect. I can see you're missing forumId, but there might be more.

Reply:
If you do this:
```suggestion
  useEffect(() => {
	    async function getSubscribe() {
	    let subscriptionRequest: SubscriptionRequest = {
	      objectType: SubscriptionObjectType.FORUM,
	      idList: [forumId],
	      sortByType: SortByType.OBJECT_ID,
	      sortDirection: Direction.ASC,
	    }
	    let subscriptionList = await SynapseClient.postSubscriptionList(
	      accessToken,
	      subscriptionRequest,
	    )
	    if (subscriptionList.totalNumberOfResults > 0) {
	      let currentSubscription = subscriptionList.results[0]
	      setSubscribed(currentSubscription)
	    }
	  }
    getSubscribe()
  }, [])
```

Or this
```tsx
async function getSubscribe(forumId: string, accessToken: string, setSubscribed: (currentSubscription: string) => void) {
	let subscriptionRequest: SubscriptionRequest = {
		objectType: SubscriptionObjectType.FORUM,
		idList: [forumId],
		sortByType: SortByType.OBJECT_ID,
		sortDirection: Direction.ASC,
	}
	let subscriptionList = await SynapseClient.postSubscriptionList(
		accessToken,
		subscriptionRequest,
	)
	if (subscriptionList.totalNumberOfResults > 0) {
		let currentSubscription = subscriptionList.results[0]
		setSubscribed(currentSubscription)
	}
}

// ...
export function ForumTable(props) {
// ...
  useEffect(() => {
    getSubscribe(forumId, accessToken, setSubscribed)
  }, [])
// ...
}
```


Then the linter will see that you omitted dependencies from the 2nd argument of useEffect. You typically want to include all dependencies unless you have a good reason for omitting them.